certbot/log.py

```diff
@@ -127,54 +171,125 @@ def format(self, record):
             return out
 
 
-def except_hook(exc_type, exc_value, trace, config):
-    """Logs exceptions and reports them to the user.
+class MemoryHandler(logging.handlers.MemoryHandler):
+    """Buffers logging messages in memory until the buffer is flushed.
+
+    This differs from logging.handlers.MemoryHandler in that flushing
+    only happens when it is done explicitly.
+
+    """
+    def __init__(self, target=None):
+        # capacity doesn't matter because should_flush() is overridden
+        capacity = float('inf')
+        if sys.version_info < (2, 7):  # pragma: no cover
+            logging.handlers.MemoryHandler.__init__(
+                self, capacity, target=target)
+        else:
+            super(MemoryHandler, self).__init__(capacity, target=target)
+
+    def shouldFlush(self, record):
+        """Should the buffer be automatically flushed?
+
+        :param logging.LogRecord record: log record to be considered
+
+        :returns: False because the buffer should never be auto-flushed
+        :rtype: bool
 
-    Config is used to determine how to display exceptions to the user. In
-    general, if config.debug is True, then the full exception and traceback is
-    shown to the user, otherwise it is suppressed. If config itself is None,
-    then the traceback and exception is attempted to be written to a logfile.
-    If this is successful, the traceback is suppressed, otherwise it is shown
-    to the user. sys.exit is always called with a nonzero status.
+        """
+        return False
```

Comment:
I mean, if we wanted to be fancy, we could auto-flush once we're in non-temp-handler-land. On the one hand, it's cleaner to not do inspection of the type of target we have. On the other hand, do we really want to be sitting there at the very end having to flush everything?

We could set a property of the object when we hand it off, and condition `shouldFlush` on that.